### PR TITLE
Added Traackr.com

### PR DESCRIPTION
**Pull request by shaeney**
HI, I am not really familiar with Git or php, being a .Net developer who uses TFS.

But I'd like to contribute

### CODE REVIEW

**Comment by Findus23** [comment]
@shaeney Could you add tests in `Tests/fixtures/bots.yml` for the new bots? (Similar to https://github.com/piwik/device-detector/commit/c219415c2ad114874adb77ad6753d1663710c6b7)

**Comment by shaeney** [comment]
Can anyone help? I am not familiar with open source/git, so dont know how to find out what I need to do to fix this?

**Comment by Findus23** [comment]
@shaeney The error travis is reporting (https://travis-ci.org/piwik/device-detector/jobs/201858723#L367) is that in https://github.com/piwik/device-detector/pull/5584/files#diff-658652a1b8930dae0035e1dee2883900R1390 you set the url of the bot to an empty string but in `Tests/fixtures/bots.yml` you are missing the `url` key. 
Best solution is to add an url (http://feeltiptop.com) to both of them.

In addition you'll need a similar test for Traackr.com and WebClient.
And it would be nice to add more information to WebClient if you have any. (Or just add it with the name "Generic Bot" if there are none)

**Comment by shaeney** [comment]
@Findus23 Can you give me a hand? While I am familiar with TDD in a .Net world, I dont really understand whats expected of me here. I've tried to reverse engineer what I need to do, but cant work it out?

I thought Id need to add text to
Tests\fixtures\bots.yml for example

- 
  user_agent: Traackr.com
  bot:
    name: Traackr.com
    category: Crawler
    url: http://www.traackr.com

and this would be matched by the equivalent in the regex bots.yml?

But I cant interpret what the error log is telling me?

**Comment by Findus23** [comment]
@shaeney The last commit is looking good. Are you sure that the complete user agent is `Traackr.com`? 
For the key `user_agent` in the test you need an example of the user agent.

File: Tests/fixtures/bots.yml [hunk 1]
```diff
@@ -2649,4 +2649,35 @@
...
+  user_agent: http://feeltiptop.com
+  bot:
+    name: feeltiptop.com
+    category: Crawler
+    producer:
+      name: feeltiptop.com
+      url: http://feeltiptop.com
+
+- 
+  user_agent: Traackr.com
```
Comment (Findus23) [inline Tests/fixtures/bots.yml]: This test is not needed any more as the second one is working

File: regexes/bots.yml [hunk 2]
```diff
@@ -1376,10 +1376,27 @@
...
 
+- regex: 'Traackr.com'
+  name: 'Traackr.com'
+  category: 'Crawler'
+  url: 'http://www.traackr.com'
+
+- regex: 'feeltiptop'
+  name: 'feeltiptop.com'
+  category: 'Crawler'
+  url: ''
```
Comment (Findus23) [inline regexes/bots.yml]: Could you add the url here instead of an empty string?

File: Tests/fixtures/bots.yml [hunk 3]
```diff
@@ -2649,4 +2649,35 @@
...
   bot:
-    name: Generic Bot
\ No newline at end of file
+    name: Generic Bot
+
+- 
+  user_agent: http://feeltiptop.com
+  bot:
+    name: feeltiptop.com
+    category: Crawler
```
Comment (Findus23) [inline Tests/fixtures/bots.yml]: You need to add the url here

File: regexes/bots.yml [hunk 4]
```diff
@@ -1376,10 +1376,27 @@
...
+  producer:
+    name: 'feeltiptop.com'
+    url: 'http://feeltiptop.com'
+
+- regex: 'WebClient/1.0'
+  name: 'WebClient'
```
Comment (Findus23) [inline regexes/bots.yml]: You need to change the name to 'Generic Bot' to match the test.

**Review by Findus23 — CHANGES_REQUESTED** [review]
I hope this helps

File: regexes/bots.yml [hunk 5]
```diff
@@ -1408,6 +1408,27 @@
...
+
+# Some libraries used for bots
+- regex: '(HTTP_Request2|Mechanize|aiohttp|Google-HTTP-Java-Client|WWW-Mechanize|faraday)'
+  name: 'Generic Bot'
```
Comment (sgiehl) [inline regexes/bots.yml]: Guess this block moved in while merging. It shouldn't be present in that file anymore, so please remove it as well.

**Comment by sgiehl** [comment]
Closing stalled pull request

**Pull request closed without merging**